picky-asn1-x509/src/private_key_info.rs

```diff
@@ -104,24 +109,46 @@ impl<'de> de::Deserialize<'de> for PrivateKeyInfo {
                 A: de::SeqAccess<'de>,
             {
                 let version = seq_next_element!(seq, PrivateKeyInfo, "version");
-                if version != 0 {
-                    return Err(serde_invalid_value!(
-                        PrivateKeyInfo,
-                        "unsupported version (valid version number: 0)",
-                        "a supported PrivateKeyInfo"
-                    ));
-                }
-
-                let private_key_algorithm: AlgorithmIdentifier =
-                    seq_next_element!(seq, PrivateKeyInfo, "private key algorithm");
-                let private_key = if private_key_algorithm.is_a(oids::rsa_encryption()) {
-                    PrivateKeyValue::RSA(seq_next_element!(seq, PrivateKeyInfo, "rsa oid"))
+
+                let tag_peek: TagPeeker = seq.next_element::<TagPeeker>()?.expect("I hope TagPeeker never panics");
+                let (private_key, private_key_algorithm) = if tag_peek.next_tag == Tag::OCTET_STRING {
+                    let private_key = PrivateKeyValue::EC(ECPrivateKey {
+                        private_key: seq_next_element!(seq, OctetStringAsn1, "elliptic curve private key"),
```

Comment:
This is a workaround to deserialize the `ECPrivateKey` fields instead of the whole struct. As the whole deserialization expects a struct's fields to be in a `SEQUENCE`. I didn't find how to make it with `serde`. @CBenoit Maybe, you can come up with a proper solution.

Reply:
I believe this code is mixing up the PKCS8 format with the Elliptic Curve Private Key Format, however, those are distinct. The PKCS8 is _wrapping_ the ECPrivateKey behind an `OctetStringAsn1Container` (like it's done for `RsaPrivateKey`).